projects/aca-content/viewer/src/lib/components/viewer/viewer.component.spec.ts

```diff
@@ -22,41 +22,332 @@
     });
 
-    store = TestBed.inject(Store);
-    spyOn(store, 'select').and.returnValue(EMPTY);
     fixture = TestBed.createComponent(AcaViewerComponent);
     component = fixture.componentInstance;
+
+    router = TestBed.inject(Router);
+    route = TestBed.inject(ActivatedRoute);
+    contentApi = TestBed.inject(ContentApiService);
+    uploadService = TestBed.inject(UploadService);
     nodesApiService = TestBed.inject(NodesApiService);
+    appHookService = TestBed.inject(AppHookService);
+    store = TestBed.inject(Store);
+  });
+
+  it('should display document upon init', () => {
+    route.params = of({
+      folderId: 'folder1',
+      nodeId: 'node1'
+    });
+    spyOn(component, 'displayNode').and.stub();
+
+    component.ngOnInit();
+
+    expect(component.folderId).toBe('folder1');
+    expect(component.displayNode).toHaveBeenCalledWith('node1');
+  });
+
+  it('should not display node when id is missing', async () => {
+    spyOn(router, 'navigate').and.stub();
+    spyOn(contentApi, 'getNodeInfo').and.returnValue(of(null));
+
+    await component.displayNode(null);
+
+    expect(contentApi.getNodeInfo).not.toHaveBeenCalled();
+    expect(router.navigate).not.toHaveBeenCalled();
+  });
+
+  it('should navigate to next and previous nodes', () => {
+    spyOn(store, 'dispatch');
+    spyOn<any>(component, 'getFileLocation').and.returnValue(fakeLocation);
+    const clickEvent = new MouseEvent('click');
+
+    component.previousNodeId = 'previous';
+    component.onNavigateBefore(clickEvent);
+    expect(store.dispatch).toHaveBeenCalledWith(new ViewNodeAction('previous', { location: fakeLocation }));
+
+    component.nextNodeId = 'next';
+    component.onNavigateNext(clickEvent);
+    expect(store.dispatch).toHaveBeenCalledWith(new ViewNodeAction('next', { location: fakeLocation }));
   });
 
-  describe('Load content', () => {
-    it('should call node update after RefreshPreviewAction is triggered', () => {
-      spyOn(nodesApiService.nodeUpdated, 'next');
-      component.ngOnInit();
-      const node = new Node();
+  it('should reload document list and navigate to correct location upon close', async () => {
+    spyOn(store, 'dispatch');
+    spyOn<any>(component, 'navigateToFileLocation').and.callThrough();
+    spyOn<any>(component, 'getFileLocation').and.returnValue(fakeLocation);
+    spyOn(router, 'navigateByUrl').and.returnValue(Promise.resolve(true));
+
+    component.onViewerVisibilityChanged();
+
+    expect(store.dispatch).toHaveBeenCalledWith(new ReloadDocumentListAction());
+    expect(component['navigateToFileLocation']).toHaveBeenCalled();
+    expect(router.navigateByUrl).toHaveBeenCalledWith(fakeLocation);
+  });
+
+  it('should navigate to original location in case of Alfresco API errors', async () => {
+    component['previewLocation'] = 'personal-files';
+    spyOn(contentApi, 'getNodeInfo').and.returnValue(throwError('error'));
+    spyOn(JSON, 'parse').and.returnValue({ error: { statusCode: '123' } });
```

Comment:
instead of mocking JSON.parse please try to prepare error object returned by getNodeInfo which gives you expected result in JSON.parse